### ISSUE TITLE
Eslint doesn't work for components outside "pages" dir

### ISSUE THREAD

**Issue by KoretzkiyBogdan** [vercel/next.js#26959]
### What version of Next.js are you using?

11.01

### What version of Node.js are you using?

14.17.1

### What browser are you using?

n/a

### What operating system are you using?

mac os

### How are you deploying your application?

n/a

### Describe the Bug

I created dir named "@core" on the same as "pages" folder level and wrote code that broke some eslint rules and I didn't receive any warns/errors when run `npm run lint`
But if I break some rule inside "pages" dir - it will show me errors as expected.
Also I can see these errors in vs code (with eslint plugin)

My eslintrc config:
`{
  "plugins": ["@typescript-eslint"],
  "extends": [
    "next",
    "plugin:@typescript-eslint/recommended",
    "plugin:prettier/recommended"
  ],
  "ignorePatterns": ["next.config.js"],
  "rules": {
    "@typescript-eslint/no-explicit-any": "error"
  }
}
`

### Expected Behavior

It will show warning and errors in the console

### To Reproduce

1. Create directory with js or ts code near "pages".
2. Include some modules from this directory to one of pages.
3. Write some code that breaks eslint rules in file which outside "pages" directory.
4. Run `npm run lint`

**Comment by dan-turner** [comment]
Experiencing the same. Also tried reverting to 11.0.0 but same result.

**Comment by Dragate** [comment]
Isn't that expected behaviour? [Docs](https://nextjs.org/docs/basic-features/eslint#linting-custom-directories) mention that ESLint runs on specific folders, which can be customized in `next.config.js` or through the `--dir` flag.

**Comment by KoretzkiyBogdan** [comment]
> Isn't that expected behaviour? [Docs](https://nextjs.org/docs/basic-features/eslint#linting-custom-directories) mention that ESLint runs on specific directories, which can be customized in `next.config.js` or through the `--dir` flag.

Yeah, you're right. Thank you.
Also I discovered bug that approach with `next.config.js` doesn't work, only with cli arguments

**Comment by housseindjirdeh** [comment]
@Dragate is correct, this is the current expected behavior. `pages/`, `components/`, and `lib/` are linted by default and you'll have to use the `--dir` flag to specify custom directories.

```
next lint --dir @core
```

Two quick things worth bringing up:

1. There's an open PR (#26777) to simplify the syntax to match that of the ESLint CLI. Once this lands, you won't need to use the `--dir` flag and can just specify folders directly (`next lint @core`). This is a breaking change however, so will most likely land by v12.
2. I can understand the confusion of how `next lint` doesn't work with directories specified in `next.config.js` and only with the `--dir` flag. There's an open PR (https://github.com/vercel/next.js/pull/26401) that fixes this so that `next lint` respects both :)

------

Closing this, but please let me know if there's anything still unresolved and I can re-open it!

**Comment by balazsorban44** [comment]
This issue has been automatically locked due to no recent activity. If you are running into a similar issue, please create a new issue with the steps to reproduce. Thank you.
